core/src/main/scala/com/gu/play/secretrotation/Phase.scala

```diff
@@ -12,9 +12,9 @@ trait Phase[T] {
 
   val alsoAccepted: Traversable[T]
 
-  final val onlyAcceptsActive = alsoAccepted.isEmpty
+  final def onlyAcceptsActive = alsoAccepted.isEmpty
 
-  final val accepted: Stream[T] = Stream(active) ++ alsoAccepted
+  final def accepted: Stream[T] = Stream(active) ++ alsoAccepted
```

Comment:
These `val`s seemed to be getting evaluated too early, causing a `NullPointerException` at `com.gu.play.secretrotation.Phase.$init$(Phase.scala:15)` as seen in the stacktrace below.

Eager evaluation of the `onlyAcceptsActive` field was triggering the `alsoAccepted` field to be evaluated _before_ it had been populated - so we were calling `.isEmpty` on `null`.

```
play.api.UnexpectedException: Unexpected exception[NullPointerException: null]
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:186)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:124)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:222)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:137)
        at akka.stream.impl.fusing.MapAsync$$anon$25.onPush(Ops.scala:1194)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:519)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:585)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:469)
Caused by: java.lang.NullPointerException: null
        at com.gu.play.secretrotation.Phase.$init$(Phase.scala:15)
        at com.gu.play.secretrotation.Phase$$anon$1.<init>(Phase.scala:22)
        at com.gu.play.secretrotation.Phase.map(Phase.scala:22)
        at com.gu.play.secretrotation.Phase.map$(Phase.scala:19)
        at com.gu.play.secretrotation.DualSecretTransition$Phase$PhaseAge.map(DualSecretTransition.scala:27)
        at com.gu.play.secretrotation.DualSecretTransition$TransitioningSecret$$anon$3.secrets(DualSecretTransition.scala:73)
        at com.gu.play.secretrotation.RotatingSecretComponents.configuration(RotatingSecretComponents.scala:17)
        at com.gu.play.secretrotation.RotatingSecretComponents.configuration$(RotatingSecretComponents.scala:16)
        at configuration.ApplicationComponents.configuration(ApplicationComponents.scala:31)
        at play.api.BuiltInComponents.actorSystem(Application.scala:326)
```